arviz/data/inference_data.py

```diff
@@ -326,18 +326,25 @@ def from_netcdf(filename: str) -> "InferenceData":
         ----------
         filename : str
             location of netcdf file
+        group_kwargs : dict of dict
+            Keyword arguments to be passed into each call of `xarray.open_dataset`.
 
         Returns
         -------
         InferenceData object
         """
         groups = {}
+
         try:
             with nc.Dataset(filename, mode="r") as data:
                 data_groups = list(data.groups)
 
             for group in data_groups:
-                with xr.open_dataset(filename, group=group) as data:
+                if group_kwargs is not None and group in group_kwargs:
+                    group_kws = group_kwargs[group]
+                else:
+                    group_kws = {}
```

Comment:
I think we could try something like

```suggestion
                group_kws = {}
                if group_kwargs is not None and regex is False:
                    group_kws = group_kwargs.get("group", {})
                if group_kwargs is not None and regex is True:
                    for key, kws in group_kwargs.items():
                        if re.search(key, group):
                            group_kws = kws
```

So that using `group_kwargs = {".+": {"dim": 3}}` will use the same kwargs everywhere or using `group_kwargs = {"posterior.*": {"dim": 3}}` will share kwargs between posterior and posterior predictive. 

Another option would be to use the functions we use for the `groups` argument in inferencedata methods, so that not only regex can be used but metagroups too. I think it could look like:

```
class _IDataGroupHolder(InferenceData):
    # ideally we would move https://github.com/arviz-devs/arviz/blob/main/arviz/data/inference_data.py#L1430
    # to be a function instead of a method and then have the method get _groups_all and call this function.
    # I hope this works to play around
    def __init__(self, groups):
        self._groups = groups

def expand_group_kwargs(groups, group_kwargs, filter_groups=None):
    ghandler = _IDataGroupHandler(groups)
    if any([key.startswith("~") for key in group_kwargs.keys()) and len(group_kwargs) > 1:
        raise ValueError("Filtering with ~ is only supported for a single key")
    extended_kwargs = {}
    for key, kwargs in group_kwargs:
        extended_keys = ghandler._group_names()
        # probably add a check to raise error if some of the values in extended_keys are already keys in extended_kwargs
        extended_kwargs.update({key: kwargs for key in extended_keys})
    return extended_kwargs

# so netcdf first calls this function, then it uses your current implementation
```

With this, the two examples above would work if using `filter_groups="regex"` and in addition, `group_kwargs = {"posterior_groups": {"dim": 3}}` would apply the chunking to posterior, sample stats, posterior predictive and log likelihood groups (only if present) as defined in https://github.com/arviz-devs/arviz/blob/main/arvizrc.template#L11. And users could always define custom metagroups so that they only need a key in group_kwargs which is a bit more verbose than using a single dict, but I think it's not too inconvenient.

A 3rd option might be configparser?

Reply:
I haven't used the regex option myself, but if it is useful, I think this could fit comfortably in this PR.
Agree with the follow up PR with configparser - I agree that getting a barebones implementation that exposes more xarray functionality is a good first step. I don't think I'll be able to touch this PR for the next couple of weeks, but I think the suggestions are reasonable and I'll aim to address them when I get around to them.